elastic/beats
# Use alias to report container image in k8s parts

## What does this PR do?
This PR leverages `type :alias`  for `kubernetes.container.image` so as to copy the fields under ECS `container.image.name`.
This affects metadata collected by kubernetes autodiscover provider, `add_kubernetes_metadata` processor and `state_container` metricset.

## Why is it important?
In order to have a persistent way to report ECS container fields, without the need to duplicate fields and store both which will impact ECS.


## How to test this PR locally
1. Before all test verify that `fields.yml` on your testing env is properly updated with the `alias` change and manually `setup` and verify that mappings are properly loaded in ES.
2. Check that autodiscover meta work:
```
filebeat.autodiscover:
      providers:
        - type: kubernetes
          node: ${NODE_NAME}
          hints.enabled: true
          hints.default_config:
            type: container
            paths:
              - /var/log/containers/*${data.kubernetes.container.id}.log
```
Make sure that `container.image.name` exist in the final documents in ES and you able to search on based on it while  `kubernetes.container.image` is not populated in the event but is searchable because of the `alias`.
3. Check that `add_kubernetes_metadata` processor works:
```
filebeat.inputs:
    - type: container
      paths:
        - /var/log/containers/*.log
      processors:
        - add_kubernetes_metadata:
            host: ${NODE_NAME}
            indexers:
            - container:
            matchers:
            - logs_path:
                logs_path: "/var/log/containers/"
```
Make sure that `container.image.name` exist in the final documents in ES and you able to search on based on it while  `kubernetes.container.image` is not populated in the event but is searchable because of the `alias`.

4. Verify with `state_container` metricset enabled that, `container.image.name` exist in the final documents in ES and you able to search based on it while  `kubernetes.container.image` is not populated in the event but is searchable because of the `alias`.


## Related issues


- Closes https://github.com/elastic/beats/issues/23585
- Relates https://github.com/elastic/beats/pull/20984, https://github.com/elastic/beats/pull/23802


## Screenshots
<img width="1045" alt="Screenshot 2021-03-10 at 12 30 38 PM" src="https://user-images.githubusercontent.com/11754898/110615899-97682c00-819c-11eb-873f-7e03bbaf8d45.png">
<img width="1045" alt="Screenshot 2021-03-10 at 12 30 28 PM" src="https://user-images.githubusercontent.com/11754898/110615918-9cc57680-819c-11eb-8110-04dccf8dcde3.png">
<img width="1609" alt="Screenshot 2021-03-10 at 12 41 07 PM" src="https://user-images.githubusercontent.com/11754898/110617232-1ad64d00-819e-11eb-8d2a-b7347a75f1f2.png">
<img width="1609" alt="Screenshot 2021-03-10 at 12 40 54 PM" src="https://user-images.githubusercontent.com/11754898/110617245-1e69d400-819e-11eb-9ffc-5a1574057f8f.png">
